src/distributions/uniform.rs

```diff
@@ -822,28 +849,29 @@ macro_rules! uniform_float_impl {
         impl UniformSampler for UniformFloat<$ty> {
             type X = $ty;
 
-            fn new<B1, B2>(low_b: B1, high_b: B2) -> Self
+            fn new<B1, B2>(low_b: B1, high_b: B2) -> Result<Self, Error>
             where
                 B1: SampleBorrow<Self::X> + Sized,
                 B2: SampleBorrow<Self::X> + Sized,
             {
                 let low = *low_b.borrow();
                 let high = *high_b.borrow();
-                debug_assert!(
-                    low.all_finite(),
-                    "Uniform::new called with `low` non-finite."
-                );
-                debug_assert!(
-                    high.all_finite(),
-                    "Uniform::new called with `high` non-finite."
-                );
-                assert!(low.all_lt(high), "Uniform::new called with `low >= high`");
+                /*
+                if !(low.all_finite()) || !(high.all_finite()) {
+                    return Err(Error::NonFinite);
+                }
+                */
+                if !(low.all_lt(high)) {
+                    return Err(Error::EmptyRange);
+                }
```

Comment:
To resolve. Do we just include the commented code? Only in `#[cfg(debug_assertions)]`?

The original intention of the extra asserts was better error messages in debug builds without extra overhead, but the overhead may not be significant.

Reply:
Done.